sdk/packages/core/src/api/getQuote.ts

```diff
@@ -2,17 +2,28 @@ import { type Address, type Hex, fromHex, zeroAddress } from 'viem'
 import { fetchJSON } from '../internal/api.js'
 import type { Environment } from '../types/config.js'
 import type { Quote, Quoteable } from '../types/quote.js'
+import type { Prettify } from '../types/utils.js'
 import { getApiUrl } from '../utils/getApiUrl.js'
 import { toJSON } from '../utils/toJSON.js'
 
-export type GetQuoteParameters = {
-  srcChainId?: number
-  destChainId: number
-  mode: 'expense' | 'deposit'
-  deposit: Quoteable
-  expense: Quoteable
-  environment?: Environment | string
-}
+export type GetQuoteParameters = Prettify<
+  {
+    srcChainId?: number
+    destChainId: number
+    environment?: Environment | string
+  } & (
+    | {
+        mode: 'deposit'
+        deposit: Prettify<Omit<Quoteable, 'amount'>>
```

Comment:
We could make the `deposit` optional here rather than having to provide an empty object
```suggestion
        deposit?: Prettify<Omit<Quoteable, 'amount'>>
```